pylint/checkers/deprecated.py

```diff
@@ -0,0 +1,36 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/master/COPYING
+
+"""Checker mixin for deprecated functionality."""
+
+import abc
+
+import astroid
+
+
+class DeprecatedMixin(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def deprecated_methods(self):
+        """Returns Iterator of deprecated methods."""
+
+    def check_deprecated_method(self, node, inferred):
+
```

Comment:
Maybe the order here could be changed to optimize the mean number of comparison / assignment we do. Probably putting the check on the type of inferred first ? Also creating a class constant for `acceptable_nodes ` so we do not assign it every time ? What do you think ?